addon/mixins/adapter-fetch.js

```diff
@@ -109,6 +109,23 @@ export function mungOptionsForFetch(_options, adapter) {
 
   return options;
 }
+/**
+ * Function checks that there is a body in the response. Otherwise JSON.parse will throw an error.
+ * If there is not, return an empty object and let Ember data handle the empty response
+ * @param {Response} response
+ * @returns {Promise}
+ */
+export function determineBodyPromise(response) {
+  let bodyPromise;
+  const contentLength = response.headers.get('content-length');
+
+  if (contentLength && parseInt(contentLength)) {
```

Comment:
I believe `Number` constructor ends up having less caveats then parseInt. 

For example, Number is less forgiving when given junk:

```js
parseInt(' 10dd1') // => 10
Number(' 10dd1') // => NaN
```

thoughts?